### ISSUE TITLE
VRTK_Interact_Grab NullReferenceException

### ISSUE THREAD

**Issue by 10FingerArmy** [thestonefox/SteamVR_Unity_Toolkit#129]
Using the interact_grab script on the controller on a default steamvr 1.1.0 camera rig will yield two  nullreferenceexceptions.

```
NullReferenceException: Object reference not set to an instance of an object
VRTK_InteractGrab.Start () (at Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_InteractGrab.cs:95)
NullReferenceException: Object reference not set to an instance of an object
VRTK_InteractGrab.Update () (at Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_InteractGrab.cs:395)
```

which is

`controllerAttachPoint = transform.GetChild(0).Find("tip").GetChild(0).GetComponent<Rigidbody>();
`

These is quite logical when looking at the order. SteamVR initializes the models lazily so the "tip" will only be created after some frames when the model is loaded. The lookup for the attachpoint should therefore also be delayed. A better way of doing it is probably to not do it once but wrap it into a "getAttachPoint" method that does the lookup ondemand if it is not done already or got reset (think of exchanging the controller model during play). 

I was curious why it works in the examples and as it seems the model hierarchy was baked into the scene. That explains the difference but will only work with the example setup and not with the pure steamvr prefab.

**Comment by thestonefox** [comment]
The reason the camera rig prefab is different is because the steam vr plugin comes with 2. The one in the prefabs folder (the one you're referring to) and the one in the test_throw example scene with the elements there in the editor (as explained in the readme of this repo).

The camera rig prefab included in this repo is the one from the test throw scene. I need to understand why valve included two in their repo before feeling comfortable removing the prefab from this repo.

In short, your issue is being caused because you're not using the prefab from this repo

**Comment by 10FingerArmy** [comment]
Thanks for the super quick response. That let's me continue to experiment :-) I have the feeling using a prefab from some test folder is probably not the way to go but indeed it would be great to understand how they envision it. I kind of suspect it was simply a shortcut for the examples to have them synchronous because they did not want to implement the delayed attachment handler I proposed above. This cannot work if the model is exchanges at runtime.

**Comment by thestonefox** [comment]
I'd love to understand some of their choices! :)

Some of their code is great and some is just broken, I wish they were more vocal about their work.

Even their last release (1.0.9) broke builds, not sure how basic checks got past them :/

**Comment by tgilmo5** [comment]
Oh was it the SteamVR release that broke the build? Is it controller related?

**Comment by thestonefox** [comment]
@tgilmo5 If you're referring to your issue when you build, it's not related to this ticket, check out the FAQ link I sent, it will solve your problem

**Comment by thestonefox** [comment]
There's a PR open in regards to this issue.

Discussion continues over there!

https://github.com/thestonefox/SteamVR_Unity_Toolkit/pull/135

**Comment by spaggi** [comment]
Hi,

I am getting the same error because I attached a child object to my Controller. Without wanting to mess with your code - I believe this should be made a bit more stable, not relying exact on the child positions.

Check the image - this setup will lead to the error

https://i.imgur.com/pcwf1kt.png

**Comment by thestonefox** [comment]
@spaggi there is a PR open to deal with this issue

**Comment by spaggi** [comment]
Perfect, thanks for letting me know!

**Comment by thestonefox** [comment]
Fixed with commit https://github.com/thestonefox/SteamVR_Unity_Toolkit/commit/aa3dcf0d90e97dd3f97b0a1de175278e0f847954
